core/src/main/java/com/spotify/metrics/core/codahale/metrics/ext/MetricRegistryListener.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.metrics.core.codahale.metrics.ext;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import com.spotify.metrics.core.DerivingMeter;
+
+public interface MetricRegistryListener extends com.codahale.metrics.MetricRegistryListener {
+    /**
+     * Called when a {@link DerivingMeter} is added to the registry.
+     *
+     * @param name          the meter's name
+     * @param derivingMeter the meter
+     */
+    void onDerivingMeterAdded(String name, DerivingMeter derivingMeter);
```

Comment:
Nope, it is not. Distribution captures the recorded data distribution. It is used to compute percentile in a distributed environment. Can one use distribution to compute rate ?  I doubt it won't be accurate since distribution is just capturing an approximation not the actual count.

Reply:
How do distributions relate to histograms? https://github.com/infusionsoft/yammer-metrics/blob/master/metrics-core/src/main/java/com/codahale/metrics/Histogram.java

Do we need to have both? Is the newly added Distribution a superset of histogram?

Reply:
So distribution is a name for the type of Histogram that computes rank related stat using actual data distribution. 
Dropwizard histogram doesn't support that feature.   For more info, take a look at this issue https://github.com/spotify/heroic/issues/476

Do we need to have both ?
At the moment we  think yes. Some users may have a preference for local histogram. And I think it is the standard in the industry to support both at least   DataDog  and  OpenTelemetry support both types.